### DISCUSSION TITLE
Source code of libbinaryen.a

### DISCUSSION THREAD

**Discussion by ubaidsk** [WebAssembly/binaryen#4595] [Q&A]
I needed to look at the source code of `libbinaryen.a`, please, could you possibly share which files get compiled to make the `libbinaryen.a` file?

**Comment by kripken** [comment]
Basically everything in `src/` gets built into that library, except for the specific tools (each is under `src/tools/..foo..`) and except for the binaryen.js integration.

For more details see the CMakeList, I think that happens here:

https://github.com/WebAssembly/binaryen/blob/8bdb2887215e7e291453d42b29da400bf864ad6c/CMakeLists.txt#L330-L337
(each of those is a subdirectory of `src/`, see CmakeLists.txt in each, but they basically include all their contents)

https://github.com/WebAssembly/binaryen/blob/8bdb2887215e7e291453d42b29da400bf864ad6c/CMakeLists.txt#L350-L357
(that is where the library is defined)

**Reply by ubaidsk** [reply]
Got it. Thank you!
